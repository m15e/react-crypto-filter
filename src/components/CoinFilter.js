import React, { useEffect } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

const CoinFilter = props => {
  const { handleFilterChange, filter } = props;

  const filters = [
    "Market Cap",
    "Dollar Value",
    "All Time High",
    "24h Price Change",
  ];

  return (
    <>
      <p>Filter Top 10 by:</p>
      <select name="coinFilter">
        <option value="All">All</option>
        {filters.map(filter => (
          <option key={filter} value={filter}>
            {filter}
          </option>
        ))}
      </select>
    </>
  );
};

export default CoinFilter;
