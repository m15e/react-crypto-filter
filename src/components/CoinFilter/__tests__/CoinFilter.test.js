import React from 'react';
import renderer from 'react-test-renderer';
import CoinFilter from '../CoinFilter';
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureMockStore();
const store = mockStore({});

it('Coin component matches snapshot', () => {
  const tree = renderer.create(<Provider store={store}><CoinFilter /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});