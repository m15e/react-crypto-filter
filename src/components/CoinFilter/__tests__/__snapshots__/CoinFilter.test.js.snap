// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin component matches snapshot 1`] = `
<div
  className="container filter-container"
>
  <div
    className="columns"
  >
    <div
      className="column"
    >
      <p>
        Filter Top 10 by:
      </p>
    </div>
    <div
      className="column"
    >
      <select
        name="coinFilter"
        onChange={[Function]}
      >
        <option
          value="Show All"
        >
          Show All
        </option>
        <option
          value="Market Cap"
        >
          Market Cap
        </option>
        <option
          value="Dollar Value"
        >
          Dollar Value
        </option>
        <option
          value="All Time High"
        >
          All Time High
        </option>
      </select>
    </div>
  </div>
</div>
`;
